dwh_design_conventions.md

```diff
@@ -0,0 +1,177 @@
+it contains the facts that a good reporter seeks.
+
+Marts are stores of one or more fact tables, related primarily by their use case
+in analysis and reporting. They could be organized by business unit, high-level 
+concept, or intended set of end-users: `marketing`, `finance`, `product`, 
+etc. Some fact tables form the basis for other ones and are used across the
+project; you might gird these keystones into a `core` folder, and `ref()` them
+at will.
+
+Each mart often contains one or more transformation flows, during which staging
+data is rolled up, aggregated, nested, or otherwise altered in a more
+significant, structured way. These intermediate models should _not_ be exposed;
+they can be ephemeral or, as materializing becomes necessary for performance
+reasons, built as tables in a cordoned-off intermediate schema. (It should be
+named something that says "I'm incomplete and not to be trusted", like `intermediate`.)
+
+Where the work of staging models is limited to cleaning and preparing, fact
+tables are the product of substantive data transformation: choosing (and reducing)
+dimensions, date-spining, executing business logic, and making informed,
+confident decisions. Staging models are your supply chain; marts are your
+department store floor.
+
+### Accessories to Data
+
+There are other kinds of SQL files that find their way into robust dbt projects.
+In some cases, they offer utilities that extend your SQL, even in BI tools;
+in other cases, they exist for specific end-users, and should be put into a
+specially-permissioned schema accordingly.
+
+In addition to `staging` and `marts`, you may find yourself with model folders
+such as:
+
+**Utils:** An `all_days` table. This is useful everywhere, though it never
+forms the basis for analysis/reporting. Consider a `utils` schema.
+
+**Lookups:** A user-mapping table, a zipcode-country table, etc. You may
+reference it at several unpredictable points throughout modeling, and maybe
+even in a BI tool. These too may find themselves at home in a `utils` schema.
+
+**Admin:** Audit logs, warehouse operations, Redshift maintenance, 
+and incremental records of the miscellaneous DDL you run to make your project
+run smoothly. These make most sense in an `admin` schema, to which only
+admins have access.
+
+**Metrics:** Precisely defined measurements taken from fact tables, directly
+conducive to time-series reporting, and tightly structured so as to allow 
+one-to-one comparison with goals and forecasting. A metrics table lives
+downstream of fact tables in your DAG and deserves special status.
+
+**Packages:** While not a model folder within your main project, packages
+that include models (like our [snowplow](https://github.com/fishtown-analytics/snowplow) 
+package) can be configured into custom schema and materialization patterns
+from `dbt_project.yml`.
+
+### Postscript
+
+These thoughts are and continue to be in-progress. Fixing the scourge of

```

Comment:
This is a bit too flowery :) Try this:

1. Producing code is a collaborative exercise, and the most important goal of this guide is to standardize the way that we work across our team. As such, the bar will be quite high for PRs to get approved that don't follow these conventions.

2. This guide, like all Fishtown Analytics standards, is a work-in-progress. As we continue to build more tooling and improve our consulting practices, we'll update this guide accordingly. If you have suggestions on how to make this guide better, submit a PR.